docs/extensions/readme.md

```diff
@@ -1173,6 +1174,113 @@ When set, specifies the header parameter to be used instead of `x-ms-client-requ
+
+**Schema**:
+
+| Field Name    | Type       | Description                                      |
+| ------------- | ---------- | ------------------------------------------------ |
+| allowedTypes  | `[string]` | **Required** An array of allowed ARM resource ID kinds. Each element represents a particular type of ARM resource which can be referred to by this `arm-id`. See [Allowed formats](#allowed-arm-id-formats) for a detailed descrition of the allowed formats. |
+
+### Allowed ARM ID formats
+Note: When reading the formats below, `{}` is left as-is in the `allowedTypes` entry. Named fields such as `{provider}` must be given concrete values in the `allowedTypes` entry.
+
+| Resource kind                   | Format                                                                      | Example `allowedTypes` entry                                                       |
+| ------------------------------- | --------------------------------------------------------------------------- | ---------------------------------------------------------------------------------- |
+| Resource in a tenant            | `/providers/{provider}/{resourceType}`                                      | `/providers/Microsoft.Capacity/reservationOrders`                                  |
+| Resource in a subscription      | `/subscriptions/{}/providers/{provider}/{resourceType}`                     | `/subscriptions/{}/providers/Microsoft.Resources/resourceGroups`                   |
+| Resource in a resource group    | `/subscriptions/{}/resourceGroups/{}/providers/{provider}/{resourceType}`   | `/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks`  |
+
+Sub-resources are specified in the same manner as their parent resource but with additional paths on the end. For example to refer to a subnet: `/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks/{}/subnets`.
+
+**Example**: An `arm-id` field with no extension
+
+```json5
+"MyExampleType": {
+  "properties": {
+    "id": {
+      "type": "string",
+      "format": "arm-id"
+    }
+  }
+}
+```
+
+**Example**: An `arm-id` field that must refer to a virtual network
+```json5
+"MyExampleType": {
+  "properties": {
+    "vnetId": {
+      "type": "string",
+      "format": "arm-id",
+      "x-ms-arm-id-details": {
+        "allowedTypes": [
+          "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks"
+        ]
+      }
+    }
+  }
+}
+```
+
+**Example (preferred)**: An `arm-id` field with no extension, referring to the common type.
+```json5
+"MyExampleType": {
+  "properties": {
+    "id": {
+      "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ArmId",
```

Comment:
what's the common type 'ArmId' for ?

Reply:
@johanste mentioned that he thought it would be useful to have a shared type in the common defs (like we do for `Resource` and a few other types today).